scheduler/src/main/java/gov/nasa/jpl/aerie/scheduler/ActivityCreationTemplate.java

```diff
@@ -242,21 +240,85 @@ private ActivityInstance createInstanceForReal(final String name, final Window w
     final var success = tnw.solveConstraints();
     if (!success) {
       logger.warn("Inconsistent temporal constraints, returning empty activity");
-      return null;
+      return Optional.empty();
     }
     final var solved = tnw.getAllData(name);
-    //select earliest start time
-    final var earliestStart = solved.start().start;
-    act.setStartTime(earliestStart);
-    if (this.parametricDur == null) {
-      //select smallest duration
-      act.setDuration(solved.duration().start);
-    } else {
-      final var computedDur = this.parametricDur.compute(Window.between(earliestStart, earliestStart));
-      if (solved.duration().contains(computedDur)) {
-        act.setDuration(computedDur);
+
+    //the domain of user/scheduling temporal constraints have been reduced with the STN,
+    //now it is time to find an assignment compatible
+    //CASE 1: activity has an uncontrollable duration
+    if(Objects.equals(this.type.getDurationType(), DurationType.uncontrollable())){
```

Comment:
Object equality _might_ work (because records are magic?), but for sealed interfaces, we prefer `instanceof`

```suggestion
    if (this.type.getDurationType() instanceof DurationType.Uncontrollable) {
```

Reply:
Done.